datastore.go

```diff
@@ -290,7 +290,11 @@ func (t *txn) Query(q dsq.Query) (dsq.Results, error) {
 	txn := t.txn
 
 	it := txn.NewIterator(opt)
-	it.Seek(prefix)
+	if len(q.SeekPrefix) > 0 {
+		it.Seek([]byte(q.SeekPrefix))
+	} else {
+		it.Seek(prefix)
+	}
```

Comment:
This should probably seek to the "later" prefix (in case the seek prefix comes before the `Prefix`).

Reply:
Actually, this all depends on the ordering. If we've reversed the order, we'd want to seek to the "earlier" prefix.

Speaking of ordering, we technically need to apply the "seek" prefix *after* the order. Badger internally handles [ordering by key (ascending or descending)](https://github.com/ipfs/go-ds-badger/pull/47/files#diff-c587b3d69dfe75cd43314ecab24f0af5R279) but we have to apply all other orderings (e.g., by value) after the fact.

So, if we aren't applying any additional orders `len(order) == 0`, we're all good. Otherwise, we should either return a not supported error or apply a naive seek after ordering.